soccer/planning/planner/PivotPathPlanner.hpp

```diff
@@ -17,11 +17,16 @@ class PivotPathPlanner : public PlannerForCommandType<PivotCommand> {
     static void createConfiguration(Configuration* cfg);
     Trajectory plan(const PlanRequest& request) override;
 
-    void reset() override { previous = Trajectory{}; }
+    void reset() override {
+        previous = Trajectory{};
+        cached_pivot_point = std::nullopt;
+    }
 
 private:
-    [[nodiscard]] bool shouldReplan(const PivotCommand& command) const;
     Trajectory previous;
+
+    std::optional<Geometry2d::Point> cached_pivot_point;
```

Comment:
A comment about when this is cached / invalidated, and why we need to cache maybe?